src/rgw/rgw_dedup_utils.h

```diff
@@ -0,0 +1,442 @@
+    uint64_t egress_slabs = 0;
+
+    uint64_t single_part_objs = 0;
+    uint64_t multipart_objs = 0;
+    uint64_t small_multipart_obj = 0;
+
+    uint64_t valid_sha256 = 0;
+    uint64_t invalid_sha256 = 0;
+
+    uint64_t default_storage_class_objs = 0;
+    uint64_t default_storage_class_objs_bytes = 0;
+
+    uint64_t non_default_storage_class_objs = 0;
+    uint64_t non_default_storage_class_objs_bytes = 0;
+
+    uint64_t ingress_failed_get_object = 0;
+    uint64_t ingress_failed_get_obj_attrs = 0;
+
+    uint64_t ingress_skip_too_small_bytes = 0;
+    uint64_t ingress_skip_too_small = 0;
+
+    uint64_t ingress_skip_too_small_64KB_bytes = 0;
+    uint64_t ingress_skip_too_small_64KB = 0;
+
+    utime_t  duration = {0, 0};
+  };
+  std::ostream& operator<<(std::ostream &out, const worker_stats_t &s);
+
+  inline void encode(const worker_stats_t& w, ceph::bufferlist& bl)
+  {
+    ENCODE_START(1, 1, bl);
+    encode(w.ingress_obj, bl);
+    encode(w.ingress_obj_bytes, bl);
+    encode(w.egress_records, bl);
+    encode(w.egress_blocks, bl);
+    encode(w.egress_slabs, bl);
+
+    encode(w.single_part_objs, bl);
+    encode(w.multipart_objs, bl);
+    encode(w.small_multipart_obj, bl);
+
+    encode(w.valid_sha256, bl);
+    encode(w.invalid_sha256, bl);
+
+    encode(w.default_storage_class_objs, bl);
+    encode(w.default_storage_class_objs_bytes, bl);
+    encode(w.non_default_storage_class_objs, bl);
+    encode(w.non_default_storage_class_objs_bytes, bl);
+
+    encode(w.ingress_failed_get_object, bl);
+    encode(w.ingress_failed_get_obj_attrs, bl);
+
+    encode(w.ingress_skip_too_small_bytes, bl);
+    encode(w.ingress_skip_too_small, bl);
+
+    encode(w.ingress_skip_too_small_64KB_bytes, bl);
+    encode(w.ingress_skip_too_small_64KB, bl);
+
+    encode(w.duration, bl);
+    ENCODE_FINISH(bl);
+  }
+
+  inline void decode(worker_stats_t& w, ceph::bufferlist::const_iterator& bl)
+  {
+    DECODE_START(1, bl);
+    decode(w.ingress_obj, bl);
+    decode(w.ingress_obj_bytes, bl);
+    decode(w.egress_records, bl);
+    decode(w.egress_blocks, bl);
+    decode(w.egress_slabs, bl);
+    decode(w.single_part_objs, bl);
+    decode(w.multipart_objs, bl);
+    decode(w.small_multipart_obj, bl);
+    decode(w.valid_sha256, bl);
+    decode(w.invalid_sha256, bl);
+    decode(w.default_storage_class_objs, bl);
+    decode(w.default_storage_class_objs_bytes, bl);
+    decode(w.non_default_storage_class_objs, bl);
+    decode(w.non_default_storage_class_objs_bytes, bl);
+    decode(w.ingress_failed_get_object, bl);
+    decode(w.ingress_failed_get_obj_attrs, bl);
+    decode(w.ingress_skip_too_small_bytes, bl);
+    decode(w.ingress_skip_too_small, bl);
+    decode(w.ingress_skip_too_small_64KB_bytes, bl);
+    decode(w.ingress_skip_too_small_64KB, bl);
+
+    decode(w.duration, bl);
+    DECODE_FINISH(bl);
+  }
+
+  struct md5_stats_t {
+    void reset() {
+      memset(&this->skipped_shared_manifest, 0, offsetof(md5_stats_t, duration));
```

Comment:
is it intentional that you skip the couple first stats?
IMO, memset is a bad mechansim to reset a struct of counters. adding a counter at the beginning/end will break the reset